[elaineshijie] iP

elaineshijie opened the pull request:
# Lemon 🆒 👾 
> This is the name of Chatbot! 
### Description for GFMP

## Chatbot uses:
* add & delete tasks
* mark the completion status of tasks
* save tasks 

Type of Tasks
1. Todo
2. Deadline
3. Event

This is the`main` method:
```
public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
    }
```

More upcoming updates:
- [x] Name the chatbot
- [x] Different task types
- [ ] Good vibez by using this chatbot

This is the ~~most~~ amazing chatbot....

Done for CS2103T: [website](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.htmlurl)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

elaineshijie committed:
Level-0 Lemon

elaineshijie committed:
Level-1 Lemon

elaineshijie committed:
Level-2 Lemon

elaineshijie committed:
Level-3 Lemon

elaineshijie committed:
Level-4 Lemon

elaineshijie committed:
Level-5 Lemon

elaineshijie committed:
Level-6 Lemon

elaineshijie committed:
A-TextUiTesting A-Enums Lemon

elaineshijie committed:
Level-7 Lemon

elaineshijie committed:
Level-7 Lemon

elaineshijie committed:
Level-8 Lemon

elaineshijie committed:
Merge remote-tracking branch 'origin/branch-Level-8'

elaineshijie committed:
Implemented OOP

elaineshijie committed:
Merge remote-tracking branch 'origin/A-MoreOOP'

elaineshijie committed:
Organise in packages

elaineshijie committed:
Merge remote-tracking branch 'origin/A-Packages'

elaineshijie committed:
Merge remote-tracking branch 'origin/add-gradle-support'

elaineshijie committed:
Add JUnit Tests

elaineshijie committed:
Merge remote-tracking branch 'origin/A-JUnit'

elaineshijie committed:
Add Find Functions

elaineshijie committed:
Merge remote-tracking branch 'origin/Level-9'

elaineshijie committed:
Merge remote-tracking branch 'origin/Level-9'

elaineshijie committed:
Merge branch 'master' of github.com:elaineshijie/ip

* 'master' of github.com:elaineshijie/ip:

elaineshijie committed:
Adhere to coding standards

elaineshijie committed:
Merge remote-tracking branch 'origin/A-Coding-Standards'

elaineshijie committed:
Added JavaDocs comments

elaineshijie committed:
Merge remote-tracking branch 'origin/A-JavaDoc'

# Conflicts:
#	src/main/java/seedu/duke/parser/Parser.java
#	src/main/java/seedu/duke/ui/Ui.java

elaineshijie committed:
Fixed issue of find

elaineshijie committed:
Merge remote-tracking branch 'origin/Fix-find'

elaineshijie committed:
changes to code after checkstyle

elaineshijie committed:
Merge remote-tracking branch 'origin/A-CheckStyle'

Kurtyjlee commented on src/main/java/seedu/duke/parser/Parser.java:
```diff
@@ -0,0 +1,134 @@
+    public static void parseTasks(String input, TaskList tasks, Storage tasksData, Ui ui) throws LemonException {
+        System.out.println("input is: " + input);
+        if (!input.equals("bye")) {
+                String commandType = input.split(" ")[0].toUpperCase();
```
Is the if indentation supposed to be 8 spaces?

Kurtyjlee commented on src/main/java/seedu/duke/Exceptions/InvalidTaskException.java:
```diff
@@ -0,0 +1,8 @@
+
+public class InvalidTaskException extends LemonException {
+    public InvalidTaskException(String message) {
+        super(":( OPPS!!! I'm sorry, but I don't know what"  + message + "means :-( \n Please add todo/ deadline / event before your task description~ or other commands like mark, unmark & delete");
```
Might want to indent or start a new line here for better readability

Kurtyjlee commented on src/main/java/seedu/duke/Tasks/Event.java:
```diff
@@ -0,0 +1,45 @@
+
+    }
+
+    public Event(String description, String start, String end, boolean isDone) throws InvalidEventException {
```
Might want to give javadocs comments here

Kurtyjlee commented on src/main/java/seedu/duke/Tasks/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+     */
+    protected LocalDate by;
+
+    public Deadline(String description, String by) throws InvalidDeadlineException {
```
Might want to give javadcos comments here

Kurtyjlee reviewed (approved):
No issues with coding standards except for a few nits. Very easy to read.

elaineshijie committed:
Implement GUI

elaineshijie committed:
Merge remote-tracking branch 'origin/Level-10'

elaineshijie committed:
Add assert statements

elaineshijie committed:
Add necessary Javadocs & changes to adhere to quality code standards.

elaineshijie committed:
Merge pull request #1 from elaineshijie/A-Assertions

Add assert statements for A-Assertions

elaineshijie committed:
Merge remote-tracking branch 'origin/master' into A-CodeQuality

# Conflicts:
#	src/main/java/seedu/duke/DialogBox.java
#	src/main/java/seedu/duke/MainWindow.java

elaineshijie committed:
Merge pull request #2 from elaineshijie/A-CodeQuality

Add necessary Javadocs & changes to adhere to quality code standards.

elaineshijie committed:
Add reschedule commands to deadline tasks.

Users cannot reschedule their deadline tasks.

A reschedule command is more user-friendly because users can change the deadline of the tasks without having to delete the task and adding it with another deadline.

elaineshijie committed:
Merge remote-tracking branch 'origin/BCD-Extension'

elaineshijie committed:
Change code to raise code quality.

elaineshijie committed:
Merge remote-tracking branch 'origin/Fix-Code-Quality'

elaineshijie committed:
Adds User Guide for Lemon

elaineshijie committed:
Changes Javadocs comments to improve code quality.

elaineshijie committed:
Merge branch 'Fix-Code-Quality'

elaineshijie committed:
Changes made to error message.

elaineshijie committed:
Changes made to long method, parseTasks in Parser

elaineshijie committed:
Merge branch 'FixCodeQuality'

elaineshijie committed:
Changes made to README

elaineshijie committed:
Changes made to build gradle to fix incompatibility issues

elaineshijie committed:
Merge remote-tracking branch 'origin/Fix-JavaFXError'
